src/map.h

```diff
@@ -538,7 +537,7 @@ class VoxelStack {
 	void Save(Saver &svr) const;
 	void Load(Loader &ldr);
 
-	Voxel *voxels;   ///< %Voxel array at this stack.
+	std::unique_ptr<Voxel[]> voxels;  ///< %Voxel array at this stack.
```

Comment:
Wouldn't a vector be more useful here, or is that a next step?

Reply:
Changed. This also allowed me to rip out a lot of copy code that duplicated and replaced the whole stack every time a new voxel was added to the top or bottom; now we can just construct the new element in-place.